contracts/XVSVault/XVSVault.sol

```diff
@@ -851,4 +848,50 @@ contract XVSVault is XVSVaultStorage, ECDSA, AccessControlledV5 {
     function _ensureNonzeroAddress(address address_) internal pure {
         require(address_ != address(0), "zero address not allowed");
     }
+
+    /**
+     * @dev Transfers the reward to the user, taking into account the rewards store
+     *   balance and the previous debt. If there are not enough rewards in the store,
+     *   transfers the available funds and records the debt amount in pendingRewardTransfers.
+     * @param rewardToken Reward token address
+     * @param userAddress User address
+     * @param amount Reward amount, in reward tokens
+     */
+    function _transferReward(address rewardToken, address userAddress, uint256 amount) internal {
+        address xvsStore_ = xvsStore;
+        uint256 storeBalance = IBEP20(rewardToken).balanceOf(xvsStore_);
+        uint256 debtDueToFailedTransfers = pendingRewardTransfers[rewardToken][userAddress];
+        uint256 fullAmount = amount.add(debtDueToFailedTransfers);
+
+        if (fullAmount < storeBalance) {
+            if (debtDueToFailedTransfers != 0) {
+                emit VaultDebtUpdated(rewardToken, userAddress, debtDueToFailedTransfers, 0);
```

Comment:
I would set `pendingRewardTransfers[rewardToken][userAddress];` to 0

Reply:
61b66fd3b8c502749a5a8b1dbe0e7cab1bcf5f06